OrbitQt/SourceCodeView.cpp

```diff
@@ -0,0 +1,49 @@
+// Copyright (c) 2020 The Orbit Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#include "SourceCodeView.h"
+
+#include "OrbitBase/Logging.h"
+
+namespace orbit_qt {
+
+void SourceCodeView::SetSourceCode(QString title, QString new_code, QString language) {
+  if (title != title_ || new_code != source_code_ || language != language_) {
+    title_ = std::move(title);
+    source_code_ = std::move(new_code);
+    language_ = std::move(language);
+
+    emit sourceCodeChanged();
+  }
+}
+
+void SourceCodeView::SetHeatmapData(QVector<int> hit_counts, QVector<double> hit_ratios,
```

Comment:
How is this data organized? Is this lines to hit_count/ratio/total_ratio?

Reply:
as discussed offline: obsolete